docs/en/ingest-management/fleet-agent-proxy-support.asciidoc

```diff
@@ -0,0 +1,250 @@
+[[fleet-agent-proxy-support]]
+= Use a proxy server with {agent} and {fleet}
+
+Your organization’s security strategy and other considerations may require you
+to use a proxy server between some components in your deployment. Support is
+available in {agent} and {fleet} for connections through HTTP Connect (HTTP 1
+only) and Socks5 proxy servers.
+
+//QUESTION: Do we want to provide a list of popular proxy servers that will work?
+
+This guide describes where a proxy server is allowed in your deployment and how
+to configure proxy settings for {agent} and {fleet}. The steps for deploying the
+proxy server itself are beyond the scope of this article.
+
+NOTE: Some environments require users to authenticate with the proxy. There are
+no explicit settings for proxy authentication in {agent} or {fleet}, except the
+ability to pass credentials in the URL or as keys/tokens in headers, as
+described later.
+
+[discrete]
+[[elastic-agent-proxy-config]]
+== Configure {agent} proxy settings
+
+Configure proxy settings for {agent} when it must connect through a proxy server
+to:
+
+* Download artifacts from `artifacts.elastic.co` for subprocesses or binary
+upgrades
+* Send data to {es}
+* Retrieve agent policies from {fleet-server}
+* Retrieve agent policies from {es} (when running as a {fleet-server})
+* Monitor services (not shown)
+
+image::images/agent-proxy-server.png[Image showing connections between {agent}, {fleet-server}, and {es}]
+
+You can configure {agent} proxy settings in:
+
+* <<host-proxy-env-vars>>
+* <<es-output-proxy-settings>>
+* <<cli-proxy-settings>>
+
+[discrete]
+[[host-proxy-env-vars]]
+=== Host environment variables
+
+Set environment variables on the host to configure default proxy settings.
+The {agent} uses host environment settings by default if no proxy settings are
+specified elsewhere. You can override host proxy settings later when you
+configure the {agent} and {fleet} settings. The following environment variables
+are available on the host:
```

Comment:
It's not clear where to put these environment variables. For instance, if Elastic Agent runs as a service in Linux they should go into `/etc/sysconfig/elastic-agent` instead of system-wide shell settings such as `/etc/bashrc`.

Reply:
@ruflin @michel-laterman Where should users put these environment variables? A user on our community slack said "This is not very generic for a standard linux package given environment variables differ between redhat and debian-based OSs. we are testing with Ubuntu so it would be VERY unusual to even have a /etc/sysconfig directory." 

It looks like these docs are stalled on input from our engineering team. CC @nimarezainia in case we need to prioritize time to get this across the line. I've seen multiple people asking for these in the past few days.

Reply:
The user also wants to know where to set these for Windows as well. They tried setting env vars in system properties but they are unsure if that is the right place.

Reply:
This Powershell sequence worked for me in Windows:

```
$environment = [string[]]@(
  "HTTPS_PROXY=https://proxy-hostname:proxy-port",
  "HTTP_PROXY=http://proxy-hostname:proxy-port"
  )

Set-ItemProperty "HKLM:SYSTEM\CurrentControlSet\Services\Elastic Agent" -Name Environment -Value $environment

Restart-Service "Elastic Agent"
```

This sets `HKLM\SYSTEM\CurrentControlSet\Services\Elastic Agent\Environment` registry key of `REG_MULTI_SZ` type.

That's similar to https://www.elastic.co/guide/en/apm/agent/dotnet/current/configuration-for-windows-services.html

Reply:
@mostlyjason, you're correct about the way rhel based and debian based packages differ. We should not be recommending a specific file for all distros. rhel based has `/etc/sysconfig`, debian i believe uses `/etc/default` (though i may be incorrect there).
Though I'm not sure if we should mention these here as they are OS specific, and "environment variable" is a more general concept.
Also there may be more then one way to set it for an OS (for instance, a linux instance running systemd can provide a dropin file as well)

Reply:
> Though I'm not sure if we should mention these here as they are OS specific, and "environment variable" is a more general concept.

We did provide an example for APM in case of Windows: https://www.elastic.co/guide/en/apm/agent/dotnet/current/configuration-for-windows-services.html

Also RPM and DEB packages ship with legacy `init.d` files which reference `/etc/sysconfig/elastic-agent` (RPM) and `/etc/default/elastic-agent` (DEB) files explicitly. If it's not the right place for systemd then perhaps we should mention `Environment` option in the service file itself? Even better, why don't we ship with examples commented out?

```
[Service]
# Environment="HTTPS_PROXY=..."
```

In case of Windows we should also explain where to put environment variables for the Endpoint.

Reply:
It doesn't feel like we've reached a consensus on where to set these environment vars (and it sounds like there's a lot of "it depends"  based on the OS, whether using systemd, etc). Can someone please distill this discussion into something I can use in the docs?  Preferably something that's been tested.

@EricDavisX My apologies if I've asked you this before, but has someone on the QA team tested the proxy setup? Maybe they can provide some insight here?

I'm pretty much blocked on moving forward until I get more info.

Reply:
@gbanasiak, I like that idea.

@dedemorton, I did the initial proxy testing for the agent, however my efforts were centred around using fleet-managed agents.

I added environment variables in `/etc/environment` - however these would not be used by any service running under systemd. Those services would load environment variables from unit files like in the example provided by @gbanasiak above.
Currently on linux the service files will also refer to the legacy environment files located in places like `/etc/sysconfig` and `/etc/default`

Reply:
Examples would be perfect in my opinion, the difficulty I had was finding any reference points, also thanks to @mostlyjason and @gbanasiak for your assistance so far :)

Reply:
@dedemorton do you have enough content with @michel-laterman comment?
https://github.com/elastic/observability-docs/pull/1239#discussion_r759515626

Reply:
@jlind23 Yes, this is great. I have one follow-up question for Michel, but I can start making updates with the info I have now. Thanks everyone for following up on this.